### PR TITLE
1.29 upgrade notes

### PR DESCRIPTION
**Pull request by kwmonroe**
- Refresh both generic and 1.29-specific upgrade pages
- Blurb about 1.29 feature gaps
- Refresh ceph page with current best practices

### CODE REVIEW

File: pages/k8s/1.29/release-notes.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ The release bundle can also be [downloaded here](https://raw.githubusercontent.c
 ### Charmed Operator Framework (Ops)
 
 We're pleased to announce the completion of the Charmed Kubernetes refactor that began
-last year. Charms have moved from the `reactive` and `pod-spec` styles to the `ops`
+last year. Core charms have moved from the `reactive` and `pod-spec` styles to the `ops`
 framework in order to enable access to common charm libraries, gain better Juju support,
```
Comment (louiseschmidtgen) [inline pages/k8s/1.29/release-notes.md:28]:
```suggestion
framework. This shift aims to enable access to common charm libraries, gain better Juju support,
```

File: pages/k8s/upgrading.md [hunk 2]
```diff
@@ -60,12 +58,12 @@ As with all upgrades, there is a possibility that there may be unforeseen diffic
...
--   You read the [Upstream release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#deprecation) for details of deprecation notices and API changes for Kubernetes 1.24 which may impact your workloads.
+-   The machine from which you will perform the backup has sufficient internet access to retrieve updated software.
+-   Your cluster is running normally.
+-   Your Juju client and controller/models are running the same, stable version of Juju (see the [Juju docs][juju-controller-upgrade]).
+-   You read the [Upgrade notes][notes] to see if any caveats apply to the versions you are upgrading to/from.
+-   You read the [Release notes][release-notes] for the version you are upgrading to, which will alert you to any important changes to the operation of your cluster.
+-   You read the [Upstream release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.29.md#deprecation) for details of Kubernetes deprecation notices and API changes that may impact your workloads.
```
Comment (evilnick) [inline pages/k8s/upgrading.md:66]: i would have removed the . from this one but :man_shrugging:

File: pages/k8s/1.29/upgrading.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ toc: False
 
 It is recommended that you keep your **Kubernetes** deployment updated to the latest available stable version. You should also update the other applications which make up **Charmed Kubernetes**. Keeping up to date ensures you have the latest bug-fixes and security patches for smooth operation of your cluster.
 
-You can check the latest release version on the [Kubernetes release page on GitHub][k8s-release].
+New minor versions of **Kubernetes** are set to release three times per year. You can check the latest release version on the [Kubernetes release page on GitHub][k8s-release].
```
Comment (evilnick) [inline pages/k8s/1.29/upgrading.md:18]: I feel we ought to call out the 'integration gaps' stuff quite prominently here

**Review by evilnick — APPROVED** [review]

**Comment by evilnick** [comment]
@kwmonroe thanks for the tweaks - see comment about the integration gaps stuff - I think for now I'd like to see a big warning at the top of the specific upgrade doc if we don't want those people to upgrade

File: pages/k8s/upgrading.md (hunk 2 above)
Comment (kwmonroe) [reply pages/k8s/upgrading.md:66]: lol, i never know when to full-stop sentences in bullet points.

File: pages/k8s/1.29/upgrading.md (hunk 3 above)
Comment (kwmonroe) [reply pages/k8s/1.29/upgrading.md:27]: i put a big warning at the very top for both new + gap areas that will affect upgrades.

File: pages/k8s/1.29/upgrading.md [hunk 4]
```diff
@@ -13,9 +13,18 @@ layout: [base, ubuntu-com]
 toc: False
 ---
 
+<div class="p-notification--caution is-inline">
+  <div markdown="1" class="p-notification__content">
+    <span class="p-notification__title">Caution:</span>
+    <p class="p-notification__message">This release includes topology changes and new best practices for integrating <strong>Charmed Kubernetes</strong> with other Juju ecosystem solutions. Be sure to read and understand the *What's new* section of the <a href="/kubernetes/docs/1.29/release-notes">1.29 release notes</a> prior to upgrading your cluster.<br/>
+    <br/>
+    Additionally, some features from previous <strong>Charmed Kubernetes</strong> releases are not yet available in this release. If you rely on a compoent identified as an *Integration gap* in the <a href="/kubernetes/docs/1.29/release-notes#notes-issues">Notes and Known Issues</a> section of the release notes, remain on release 1.28 (or earlier) and do not upgrade to 1.29 at this time.</p>
```
Comment (evilnick) [inline pages/k8s/1.29/upgrading.md:21]:
```suggestion
    Additionally, some features from previous <strong>Charmed Kubernetes</strong> releases are not yet available in this release. If you rely on a component identified as an *Integration gap* in the <a href="/kubernetes/docs/1.29/release-notes#notes-issues">Notes and Known Issues</a> section of the release notes, remain on release 1.28 (or earlier) and do not upgrade to 1.29 at this time.</p>
```

Comment (evilnick) [reply pages/k8s/1.29/upgrading.md:21]: I knew there would be a typo somewhere...

**Comment by evilnick** [comment]
Thanks for the updates. I'm going to merge this now because there are some things I want to get to the bottom of

**Pull request merged**